relay-common/src/glob3.rs

```diff
@@ -12,6 +12,50 @@ fn is_match(globs: &[Regex], message: &[u8]) -> bool {
     globs.iter().any(|regex| regex.is_match(message.as_ref()))
 }
 
+/// A singular glob-pattern.
+#[derive(Debug, Clone, PartialEq, Default)]
+pub struct GlobPattern(GlobPatterns);
+
+impl GlobPattern {
+    /// Returns a new instance of [`GlobPattern`].
+    pub fn new(pattern: impl Into<String>) -> Self {
+        Self(GlobPatterns::new(vec![pattern.into()]))
+    }
+
+    /// Returns `true` if the pattern match the given message.
+    pub fn is_match<S>(&self, message: S) -> bool
+    where
+        S: AsRef<[u8]>,
+    {
+        self.0.is_match(message)
+    }
+}
+
+impl Serialize for GlobPattern {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        debug_assert_eq!(self.0.patterns.len(), 1);
+
+        match self.0.patterns.first() {
```

Comment:
I would take the panic here, the newtype was made with exactly that constraint, that is will always contain exactly one pattern:

```rs
serializer.serialize_str(&self.0.patterns[0])
```

Should discuss this with the team though or just ignore this suggestion.